### ISSUE TITLE
SciPy v1.2.0rc1 cannot be imported on Python 2.7.15

### ISSUE THREAD

**Issue by 1313e** [scipy/scipy#9540]
My title says it all, basically.
When I try to import `scipy.stats` on Python 2.7.15 using SciPy v1.2.0rc1 (which, apparently, is on PyPI despite it being a development version), I get the following error (which was generated by Travis CI):
```
import scipy.stats
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/stats/__init__.py:367: in <module>
    from .stats import *
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/stats/stats.py:173: in <module>
    from . import distributions
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/stats/distributions.py:10: in <module>
    from ._distn_infrastructure import (entropy, rv_discrete, rv_continuous,
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/stats/_distn_infrastructure.py:16: in <module>
    from scipy.misc import doccer
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/misc/__init__.py:68: in <module>
    from scipy.interpolate._pade import pade as _pade
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/interpolate/__init__.py:175: in <module>
    from .interpolate import *
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/interpolate/interpolate.py:32: in <module>
    from .interpnd import _ndim_coords_from_arrays
interpnd.pyx:1: in init scipy.interpolate.interpnd
    ???
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/spatial/__init__.py:104: in <module>
    from . import transform
../../../virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/spatial/transform/__init__.py:22: in <module>
    from .rotation import Rotation, Slerp
E     File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/spatial/transform/rotation.py", line 1574
E   SyntaxError: Non-ASCII character '\xe2' in file /home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/scipy/spatial/transform/rotation.py on line 1575, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

If I attempt to do the same on Python 3.5/3.6/3.7, it works perfectly fine.
I'm using NumPy 1.15.4 and the Xenial distribution on Travis CI.

**Comment by rgommers** [comment]
> using SciPy v1.2.0rc1 (which, apparently, is on PyPI despite it being a development version)

You should only be getting that when you do `pip install scipy --pre` (to select pre-releases)

**Comment by rgommers** [comment]
the error is odd, that should have been caught by CI if it's a real issue

**Comment by 1313e** [comment]
> > using SciPy v1.2.0rc1 (which, apparently, is on PyPI despite it being a development version)
> 
> You should only be getting that when you do `pip install scipy --pre` (to select pre-releases)

Yeah, that is what I thought as well, but I simply use `pip install -r requirements.txt`, which lists `scipy>=1.0.0`.

**Comment by rgommers** [comment]
what `pip` version did you use?

**Comment by rgommers** [comment]
I cannot reproduce that:
```
$ pip --version
pip 18.1 from /Users/rgommers/anaconda3/envs/pip-pre/lib/python3.6/site-packages/pip (python 3.6)

$ cat requirements.txt 
scipy>=1.0.0

$ pip install -r requirements.txt
Collecting scipy>=1.0.0 (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/a0/b6/70bf61c1badb5fea82d4c558e05e76c2dee5e77bb072fe465d7c7a87287d/scipy-1.1.0-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Requirement already satisfied: numpy>=1.8.2 in /Users/rgommers/anaconda3/envs/pip-pre/lib/python3.6/site-packages (from scipy>=1.0.0->-r requirements.txt (line 1)) (1.15.4)
Installing collected packages: scipy
Successfully installed scipy-1.1.0
```

And on https://pypi.org/project/scipy/#history 1.2.0rc1 is correctly labelled as a pre-release. 

My `pip` is the one packaged by `conda` though - maybe it's not 100% identical to yours. How did you install `pip`?

**Comment by 1313e** [comment]
> My `pip` is the one packaged by `conda` though - maybe it's not 100% identical to yours. How did you install `pip`?
`python -m pip install --disable-pip-version-check --user --upgrade pip`.
However, I am using Conda as well.

But, installing SciPy is not the problem, since that succeeds.
The problem is trying to import `scipy.stats`.

**Comment by rgommers** [comment]
Well, there are 2 problems. Somehow you're getting the wrong version, and then you get an exception that shouldn't be there.

**Comment by 1313e** [comment]
> Well, there are 2 problems. Somehow you're getting the wrong version, and then you get an exception that shouldn't be there.

Oh, wait, hold on.
SciPy is installed by the execution of `python setup.py install`, where `'scipy>=1.0.0'` is listed in the `install_requires` argument in the `setup` function.
It does not use the `pip install -r requirements.txt` method.

**Comment by rgommers** [comment]
`python setup.py install` where exactly?

**Comment by 1313e** [comment]
> `python setup.py install` where exactly?

Inside the directory of the package that I am trying to install.
However, I think that automatically uses `easy-install` if I am not mistaken, since the installation process generates:
```
Searching for scipy>=1.0.0
Reading https://pypi.org/simple/scipy/
Downloading https://files.pythonhosted.org/packages/44/9b/4df7879d580f90f6a9841f16a737624819b83048ba5f0c3a3d905268f836/scipy-1.2.0rc1-cp27-cp27mu-manylinux1_x86_64.whl#sha256=38857eb49f7e38d3ec079772225a79235d0bd847e24d2fa8c9a9fa70ee69f0a2
Best match: scipy 1.2.0rc1
Processing scipy-1.2.0rc1-cp27-cp27mu-manylinux1_x86_64.whl
Installing scipy-1.2.0rc1-cp27-cp27mu-manylinux1_x86_64.whl to /home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages
writing requirements to /home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg/EGG-INFO/requires.txt
Adding scipy 1.2.0rc1 to easy-install.pth file
Installed /home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/scipy-1.2.0rc1-py2.7-linux-x86_64.egg
```

**Comment by rgommers** [comment]
> However, I think that automatically uses easy-install if I am not mistaken, since the installation process generates:

Ah yes, good catch. Well, that can't be helped then. The advice you'll get from packaging people is simply to not use `python setup.py install` but `pip install .` instead.

> it used a cached SciPy build. That might make a difference.

No that shouldn't matter - `pip` first determines the version it wants, and then checks its cache.

**Comment by rgommers** [comment]
The bug is real, can reproduce. It seems the tests aren't run for `spatial.transform` on TravisCI, or that should have failed.

**Comment by 1313e** [comment]
> > However, I think that automatically uses easy-install if I am not mistaken, since the installation process generates:
> 
> Ah yes, good catch. Well, that can't be helped then. The advice you'll get from packaging people is simply to not use `python setup.py install` but `pip install .` instead.

Ah, yes, I have heard it before, but I am so used to using `python setup.py install` and never had any problems with that at all (until now I guess) that I never changed my ways of using it.
Guess now is the time to do that.

**Comment by WarrenWeckesser** [comment]
@rgommers wrote
> The bug is real,

Ralf, how did you reproduce the bug?  With Mac OS X 10.9 running Python 2.7.15, I didn't get the error with either scipy master or with 1.2.0rc1 installed with pip.

It looks like the problem is the non-ascii opening and closing double-quotes around the title of a reference.

**Comment by rgommers** [comment]
I don't have a full py27 install at hand, but a basic py27 installed with conda and then

    cd scipy/spatial/transform
    python2.7 rotation.py

shows me the `SyntaxError`.

**Comment by rgommers** [comment]
> It looks like the problem is the non-ascii opening and closing double-quotes around the title of a reference.

Indeed, fix in gh-9541. Still need to investigate why that's showing up only now.

**Comment by rgommers** [comment]
Could have to do with import order perhaps, or be platform dependent. The issue shows up for @1313e when he does `import scipy.stats`. The tests never do that. There's `# -*- coding: utf-8 -*-` comments scattered around in various files. The encoding should be set per file according to PEP 263, but perhaps that's not the case?

**Comment by 1313e** [comment]
> Could have to do with import order perhaps, or be platform dependent. The issue shows up for @1313e when he does `import scipy.stats`. The tests never do that. There's `# -*- coding: utf-8 -*-` comments scattered around in various files. The encoding should be set per file according to PEP 263, but perhaps that's not the case?

In my own package, I always use `# -*- coding: utf-8 -*-` comments everywhere as well.
The `import scipy.stats` is however done by an external package (`mlxtend`), which does not do that.
I obviously did not really know that that might be the issue, so I came here.

**Comment by WarrenWeckesser** [comment]
I don't get the syntax error with `import stats`:

```
~$ ~/miniconda2/bin/python
Python 2.7.15 |Anaconda, Inc.| (default, Nov 13 2018, 17:07:45) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import scipy
>>> scipy.__version__
'1.2.0rc1'
>>> import scipy.stats
>>> 
>>> from scipy.spatial.transform.rotation import Rotation
>>> 
```

**Comment by rgommers** [comment]
@WarrenWeckesser do you get the error with `python2.7 rotation.py`?

**Comment by crizzy9** [comment]
Has anyone solved this yet? I am still getting this error while using with tensorflow

**Comment by rgommers** [comment]
@crizzy9 it's solved in scipy 1.2.1, please upgrade to that version
